### PR TITLE
Fix #66 Openshift setup Vagrantfile for ADB

### LINKED ISSUE #66: We need some way to create an OpenShift setup for Atomic App
We should be able to create an OpenShift instance so that it can be used by Atomic Apps to run.

For example wordpress-centos7-atomicapp [1] should be able to run on OpenShift environment provided by adb.

[1] https://github.com/projectatomic/nulecule/tree/master/examples/wordpress-centos7-atomicapp

### PR DESCRIPTION
**Pull request by praveenkumar**
This vagrant file will configure openshift to adb base box.
- Get latest origin image from docker hub
- Create required configuration directory to box
- Set required permission to work with SELinux
- Create openshift router
- Create openshift default templates
- Configure Users

### CODE REVIEW

File: components/centos/centos-with-openshift/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+# The Docker registry from where we pull the OpenShift Docker image
+DOCKER_REGISTRY="docker.io"
+
+# The name of the OpenShift image.
+ORIGIN_IMAGE_NAME="openshift/origin"
+
+# The public IP address the VM created by Vagrant will get.
+# You will use this IP address to connect to OpenShift web console.
+PUBLIC_ADDRESS="10.1.2.2"
+PUBLIC_HOST="cdk.cluster.io"
```
Comment (LalatenduMohanty) [inline components/centos/centos-with-openshift/Vagrantfile]: we should s/cdk/adb/. Just a cosmetic change

File: components/centos/centos-with-openshift/README.rst [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  sure required directory mounted and *host* network is used. Wait for around 15
+  seconds to start it.
+- Check if started as expected otherwise provide docker logs for origin.
+- Make sure **oc and oadm** binaries are available to host system.
+- Create docker registry to make sure when *oc build* run it upload local images
+  to this registry.
+- Configure router so that new-app will access from web using those routes.
+- Get default templates and configure it.
+- Create *test-admin user and test project* for experiment.
+- Provide required configuration details for openshift web console and API
```
Comment (LalatenduMohanty) [inline components/centos/centos-with-openshift/README.rst]: @praveenkumar can you please put pointers in this document which will point users for doing further stuff. may be docs.openshift.org/<somepage>

**Comment by LalatenduMohanty** [comment]
LGTM

File: components/centos/centos-with-openshift/README.rst [hunk 3]
```diff
@@ -0,0 +1,25 @@
+Openshift Vagrantfile
+=====================
+
+This vagrant file is use to set openshift environment to base adb vagrant box.
```
Comment (navidshaikh) [inline components/centos/centos-with-openshift/README.rst]: *This Vagrantfile is used to set OpenShift environment to base ADB Vagrant box.

File: components/centos/centos-with-openshift/README.rst [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+During setup below steps are executed.
+
+- Create a private network and set IP *10.1.2.2*, if you want a different one
+  then change **PUBLIC_ADDRESS** variable.
+- Pull latest *openshift/origin* container from docker hub and tag it.
+- Create required configuration file directories which openshift use and set
+  SELinux flag to make sure it work with *Enforcing* mode.
+- Run docker container using origin image with different run options to make
+  sure required directory mounted and *host* network is used. Wait for around 15
```
Comment (navidshaikh) [inline components/centos/centos-with-openshift/README.rst]: *directory is mounted

File: components/centos/centos-with-openshift/README.rst [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  seconds to start it.
+- Check if started as expected otherwise provide docker logs for origin.
```
Comment (navidshaikh) [inline components/centos/centos-with-openshift/README.rst]: *Check if container has started as expected

File: components/centos/centos-with-openshift/README.rst [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+- Make sure **oc and oadm** binaries are available to host system.
+- Create docker registry to make sure when *oc build* run it upload local images
```
Comment (navidshaikh) [inline components/centos/centos-with-openshift/README.rst]: *run, it uploads

File: components/centos/centos-with-openshift/README.rst [hunk 7]
```diff
@@ -0,0 +1,25 @@
+Openshift Vagrantfile
+=====================
+
+This vagrant file is use to set openshift environment to base adb vagrant box.
+During setup below steps are executed.
```
Comment (navidshaikh) [inline components/centos/centos-with-openshift/README.rst]: I suggest we rephrase it to 
Following steps are executed as part of setup:

**Comment by LalatenduMohanty** [comment]
@praveenkumar Can you please put some description in the README  about what we are trying to achieve with the Vagrantfile. Also add a quick-start section. Basically the idea is, we should be able to just point to the README in the release notes.

File: components/centos/centos-with-openshift/README.rst [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+We want to provide a seamless experience for Openshift developer/User. This
+vagrant file take care of running and provissing Openshift in a VM with a simple
+execution of *vagrant up*. 
+
+Following steps are executed as part of provisioning:
+
+- Create a private network and set IP *10.1.2.2*, if you want a different one
+  then change **PUBLIC_ADDRESS** variable.
+- Pull latest *openshift/origin* container from docker hub and tag it.
+- Create required configuration file directories which openshift use and set
```
Comment (LalatenduMohanty) [inline components/centos/centos-with-openshift/README.rst]: s/openshift/OpenShift/

File: components/centos/centos-with-openshift/README.rst [hunk 9]
```diff
@@ -0,0 +1,88 @@
+Openshift Vagrantfile
+=====================
+
+Why Vagrantfile?
+----------------
+
+We want to provide a seamless experience for Openshift developer/User. This
+vagrant file take care of running and provissing Openshift in a VM with a simple
```
Comment (LalatenduMohanty) [inline components/centos/centos-with-openshift/README.rst]: s/OpenShift/OpenShift Origin/

File: components/centos/centos-with-openshift/README.rst [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+execution of *vagrant up*. 
+
+Following steps are executed as part of provisioning:
+
+- Create a private network and set IP *10.1.2.2*, if you want a different one
+  then change **PUBLIC_ADDRESS** variable.
+- Pull latest *openshift/origin* container from docker hub and tag it.
+- Create required configuration file directories which openshift use and set
+  SELinux flag to make sure it work with *Enforcing* mode.
```
Comment (LalatenduMohanty) [inline components/centos/centos-with-openshift/README.rst:17]: s/SELinux/Set SELinux/

File: components/centos/centos-with-openshift/README.rst [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+- Run docker container using origin image with different run options to make
+  sure required directory is mounted and *host* network is used. Wait for around 15
+  seconds to start it.
```
Comment (LalatenduMohanty) [inline components/centos/centos-with-openshift/README.rst]: s/Wait for around 15 seconds to start it/It takes around 15 seconds to start/

File: components/centos/centos-with-openshift/README.rst [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+- Check if contanter has started as expected otherwise provide docker logs for origin.
```
Comment (LalatenduMohanty) [inline components/centos/centos-with-openshift/README.rst]: s/for origin//

File: components/centos/centos-with-openshift/README.rst [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+- Make sure **oc and oadm** binaries are available to host system.
+- Create docker registry to make sure when *oc build* run, it upload local images
+  to this registry.
+- Configure router so that new-app will access from web using those routes.
+- Get default templates and configure it.
+- Create *test-admin user and test project* for experiment.
+- Provide required configuration details for openshift web console and API
```
Comment (LalatenduMohanty) [inline components/centos/centos-with-openshift/README.rst]: s/openshift/OpenShift/

File: components/centos/centos-with-openshift/README.rst [hunk 14]
```diff
@@ -0,0 +1,88 @@
...
+
+For More info about Openshift, please checkout `offical documents
```
Comment (LalatenduMohanty) [inline components/centos/centos-with-openshift/README.rst]: s/checkout/refer/

**Comment by LalatenduMohanty** [comment]
@praveenkumar looks good apart from some minor suggestions . Thanks

File: components/centos/centos-with-openshift/README.rst (hunk 10 above)
Comment (praveenkumar) [inline components/centos/centos-with-openshift/README.rst:17]: it's already 'set SELinux'

**Comment by LalatenduMohanty** [comment]
@praveenkumar LGTM

File: components/centos/centos-with-openshift/README.rst [hunk 15]
```diff
@@ -0,0 +1,88 @@
+Openshift Vagrantfile
+=====================
+
+Why Vagrantfile?
+----------------
+
+We want to provide a seamless experience for Openshift developer/User. This
+vagrant file take care of running and provissing Openshift Origin in a VM with a simple
```
Comment (navidshaikh) [inline components/centos/centos-with-openshift/README.rst]: *Vagrantfile takes care of running [..]

File: components/centos/centos-with-openshift/README.rst [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+  then change **PUBLIC_ADDRESS** variable.
+- Pull latest *openshift/origin* container from docker hub and tag it.
+- Create required configuration file directories which OpenShift use and set
+  SELinux flag to make sure it work with *Enforcing* mode.
+- Run docker container using origin image with different run options to make
+  sure required directory is mounted and *host* network is used. It takes around
+  15 seconds to start.
+- Check if contanter has started as expected otherwise provide docker logs.
+- Make sure **oc and oadm** binaries are available to host system.
+- Create docker registry to make sure when *oc build* run, it upload local images
```
Comment (navidshaikh) [inline components/centos/centos-with-openshift/README.rst]: *it uploads local images [..]

File: components/centos/centos-with-openshift/README.rst [hunk 17]
```diff
@@ -0,0 +1,88 @@
...
+- Create *test-admin user and test project* for experiment.
+- Provide required configuration details for OpenShift web console and API
+
+For More info about Openshift, please refer to `offical documents
+<https://docs.openshift.org/latest/welcome/index.html>`_.
+
+Quick Start
+-----------
+
+- Get latest adb box and add it to vagrant
```
Comment (navidshaikh) [inline components/centos/centos-with-openshift/README.rst]: *ADB box

File: components/centos/centos-with-openshift/README.rst [hunk 18]
```diff
@@ -0,0 +1,88 @@
...
+
+  ::
+  
+    $ vagrant box add adb <path_of_download_box>
+
+- Create a separate Directory for OpenShift experiment
```
Comment (navidshaikh) [inline components/centos/centos-with-openshift/README.rst]: *directory

File: components/centos/centos-with-openshift/README.rst [hunk 19]
```diff
@@ -0,0 +1,88 @@
...
+
+    $ oc status
+
+- Check available templates
+
+  ::
+
+    $ oc get templates -n openshift
+
+- Deploy a application using available templates
```
Comment (navidshaikh) [inline components/centos/centos-with-openshift/README.rst]: *Deploy an application

**Comment by navidshaikh** [comment]
LGTM
